src/stream/src/executor/source/fetch_executor.rs

```diff
@@ -101,13 +102,23 @@ impl<S: StateStore> FsFetchExecutor<S> {
                 )
```

Comment:
Just a note and may not be related to this PR: we always iterate the vnode sequentially in ascending order. Will it cause starvation? Assume we have 2 vnodes and lister will continously output newly added files evenly to each vnode. If fetcher cannot keep up with lister, fetcher may always fetch files from vnode1 and never fetch files from vnode2 but vnode2 may have older files than vnode1. cc @tabVersion

Reply:
Another note and just to make sure this is expected: under the current implementation, it is possible that for fetcher to read a partial file and skip to another file after recovery. Sequence of events:
1. lister output file `bob` (assume it is under vnode1) and fetcher puts it in the state table
2. fetcher reads some contents but not all from `boo`
3. lister output file `alice` (assume it is under vnode1) and fetcher puts it in the state table
4. recovery happens
5. fetcher restores its state via `replace_with_new_batch_reader`
6. Because `alice` has a smaller key in the state table, fetcher will fetch `alice` first before continue reading from `boo`.

Reply:
Discussed offline with @tabVersion , these are expected behaviors and we don't have any ordering guarantee for records across files.